### PR TITLE
fixed an issue with setting a boolean option to string

### PR DESCRIPTION
**Pull request by llali**
In connection details boolean options can be set to string "True" or "False" but when parsing back the value should be converted to boolean correctly

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ConnectionDetails.cs [hunk 1]
```diff
@@ -446,9 +446,16 @@ private T GetOptionValue<T>(string name)
             if (Options != null && Options.ContainsKey(name))
             {
                 object value = Options[name];
-                if(value != null && ( typeof(T) == typeof(int) || typeof(T) == typeof(int?)))
+                if (value != null && (typeof(T) != value.GetType()))
                 {
-                    value = System.Convert.ToInt32(value);
+                    if (typeof(T) == typeof(int) || typeof(T) == typeof(int?))
```
Comment (benrr101) [inline src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ConnectionDetails.cs]: Would it work if we just used Convert.ChangeType ? This way we can add a lot of type support without a lot of code additions

File: src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ConnectionDetails.cs [hunk 2]
```diff
@@ -446,9 +446,16 @@ private T GetOptionValue<T>(string name)
...
+                    {
+                        value = System.Convert.ToInt32(value);
```
Comment (benrr101) [inline src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ConnectionDetails.cs]: The use of `System` is unnecessary

File: src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ConnectionDetails.cs (hunk 1 above)
Comment (llali) [inline src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ConnectionDetails.cs]: ChangeType was not working to specifically converting to int32

Comment (benrr101) [inline src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ConnectionDetails.cs]: I'm not sure how, convert.changetype takes the type as it's second parameter, so you should be able to specify `int` as the type to convert to.

Comment (llali) [inline src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ConnectionDetails.cs]: it cannot convert logn to int, it fails, I tried that already

**Pull request merged**